lib/datadog/opentelemetry/sdk/span_processor.rb

```diff
@@ -85,6 +85,10 @@ def start_datadog_span(span)
           name, kwargs = span_arguments(span, attributes)
 
           datadog_span = Tracing.trace(name, **kwargs)
+          # The Datadog span must have the same ID as the OpenTelemetry span
+          # DEV: We need to set the span ID after the span is created
+          datadog_span.id = span.context.hex_span_id.to_i(16)
+          datadog_span.trace_id = span.context.hex_trace_id.to_i(16)
```

Comment:
Fix #1: Ensure otel and datadog spans have the same trace and span id. This will ensure `otel_span.context` returns valid ids

Reply:
Note - This approach is very naive. Not sure if it introduces race conditions or breaks span activation.

Reply:
Setting the `trace_id` is what
```ruby
          if parent_context.trace
            Tracing.send(:tracer).send(:call_context).activate!(parent_context.ensure_trace)
          else
            Tracing.continue_trace!(nil)
          end
```
does above. Any changes to the `trace_id` have to be done on the active trace.

For `span_id`, there's no API today, so I suggest adding a `_span_id` argument to `trace`, so that we can set it.

Reply:
I'll pass in the trace_id of the otel span to Tracing.continue_trace!(nil) via trace_digest